File: 20260524/exos/05_for_if.py
# ============================================================
# HOI4 — Provinces capturées
# ============================================================
# Tu repasses en revue une liste de provinces après une offensive.
# Certaines sont déjà sous ton contrôle (statut "captured").
# Compte combien tu en as prises.
#
# Lance : python3 05_for_if.py
# ============================================================

provinces = [
    {"nom": "Lille", "statut": "captured"},
    {"nom": "Arras", "statut": "enemy"},
    {"nom": "Amiens", "statut": "captured"},
    {"nom": "Rouen", "statut": "captured"},
    {"nom": "Reims", "statut": "enemy"},
]

nb_capturees = 0

# Parcours provinces avec for.
# Si le statut de la province vaut "captured", ajoute 1 à nb_capturees.
#
# Lille   → captured  →  nb_capturees = 1
# Arras   → enemy     →  nb_capturees = 1
# Amiens  → captured  →  nb_capturees = 2
# Rouen   → captured  →  nb_capturees = 3
# Reims   → enemy     →  nb_capturees = 3
#
# Résultat attendu : nb_capturees == 3
#
# Indice : for p in provinces:  puis  if p["statut"] == "captured":

# À toi :
for p in provinces:
    if p["statut"] == "captured":
        nb_capturees += 1

# --- Vérification (ne pas modifier) ---
assert nb_capturees == 3, f"Obtenu : {nb_capturees}, attendu : 3"
print("✅ Correct !")

File: 20260723/exos/05_for_if.lua
-- ============================================================
-- Hearts of Iron IV — divisions prêtes
-- ============================================================
-- Compte combien de divisions ont une orga strictement
-- supérieure à 60.
--
-- Lance : lua 05_for_if.lua
-- ============================================================

orgas = {45, 72, 61, 30, 88}
pretes = 0

-- Parcours orgas. Si x > 60, ajoute 1 à pretes.
--
-- 45 > 60 ? non
-- 72 > 60 ? oui  →  pretes = 1
-- 61 > 60 ? oui  →  pretes = 2
-- 30 > 60 ? non
-- 88 > 60 ? oui  →  pretes = 3
--
-- Résultat attendu : pretes == 3
--
-- Indice : for _, x in ipairs(...) do  +  if x > 60 then
--          Compare x, pas la table entière

-- À toi :
for i, x in ipairs(orgas) do
    if x > 60 then
        pretes = pretes + 1
    end
end

-- --- Vérification (ne pas modifier) ---
assert(pretes == 3, "Obtenu : " .. pretes .. ", attendu : 3")
print("✅ Correct !")
